app/models/station.rb

```diff
@@ -1,3 +1,71 @@
 class Station < ActiveRecord::Base
-  validates :name, :dock_count, :city, :installation_date, presence: true
+  belongs_to :city
+  validates :name,
+            :dock_count,
+            :city_id,
+            :installation_date,
+            presence: true
+
+  def self.record(station_information)
+    self.find_or_create_by(name:       station_information[:name],
```

Comment:
similar to what is happening in the city class, this is using find_by and create_ to populate the station model. again, it accepts the :city_name and using that in another method written a few lines below, it finds the :city_id in the city table.